From: Jeremy Bicha <jbicha@debian.org>
Date: Thu, 10 Mar 2022 07:52:19 -0500
Subject: Revert "status/bluetooth: Use BlueZ state to determine whether
 bluetooth is on"

This reverts commit 109e2968e294eb31c67ef0a51d6d5a56d1aa5e40.

Ubuntu 22.04 LTS isn't using the new gnome-bluetooth series
---
 js/ui/status/bluetooth.js | 21 +++++++--------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/js/ui/status/bluetooth.js b/js/ui/status/bluetooth.js
index 05050d9..2ccca5e 100644
--- a/js/ui/status/bluetooth.js
+++ b/js/ui/status/bluetooth.js
@@ -25,9 +25,7 @@ class Indicator extends PanelMenu.SystemIndicator {
         this._indicator = this._addIndicator();
         this._indicator.icon_name = 'bluetooth-active-symbolic';
         this._hadSetupDevices = global.settings.get_boolean(HAD_BLUETOOTH_DEVICES_SETUP);
-
         this._client = new GnomeBluetooth.Client();
-        this._client.connect('notify::default-adapter-powered', this._sync.bind(this));
 
         this._proxy = new RfkillManagerProxy(Gio.DBus.session, BUS_NAME, OBJECT_PATH,
                                              (proxy, error) => {
@@ -44,12 +42,9 @@ class Indicator extends PanelMenu.SystemIndicator {
 
         this._toggleItem = new PopupMenu.PopupMenuItem('');
         this._toggleItem.connect('activate', () => {
-            if (!this._client.default_adapter_powered) {
-                this._proxy.BluetoothAirplaneMode = false;
+            this._proxy.BluetoothAirplaneMode = !this._proxy.BluetoothAirplaneMode;
+            if (!this._proxy.BluetoothAirplaneMode)
                 this._client.default_adapter_powered = true;
-            } else {
-                this._proxy.BluetoothAirplaneMode = true;
-            }
         });
         this._item.menu.addMenuItem(this._toggleItem);
 
@@ -135,14 +130,12 @@ class Indicator extends PanelMenu.SystemIndicator {
         this.menu.setSensitive(sensitive);
         this._indicator.visible = nConnectedDevices > 0;
 
-        const adapterPowered = this._client.default_adapter_powered;
-
         // Remember if there were setup devices and show the menu
         // if we've seen setup devices and we're not hard blocked
         if (this._hadSetupDevices)
             this._item.visible = !this._proxy.BluetoothHardwareAirplaneMode;
         else
-            this._item.visible = adapterPowered;
+            this._item.visible = this._proxy.BluetoothHasAirplaneMode && !this._proxy.BluetoothAirplaneMode;
 
         this._item.icon.icon_name = adapterPowered
             ? 'bluetooth-active-symbolic' : 'bluetooth-disabled-symbolic';
@@ -152,11 +145,11 @@ class Indicator extends PanelMenu.SystemIndicator {
             this._item.label.text = ngettext('%d Connected', '%d Connected', nConnectedDevices).format(nConnectedDevices);
         else if (nConnectedDevices === 1)
             this._item.label.text = connectedDevices[0].name;
-        else if (adapterPowered)
-            this._item.label.text = _('Bluetooth On');
-        else
+        else if (this._adapter === null)
             this._item.label.text = _('Bluetooth Off');
+        else
+            this._item.label.text = _('Bluetooth On');
 
-        this._toggleItem.label.text = adapterPowered ? _('Turn Off') : _('Turn On');
+        this._toggleItem.label.text = this._client.default_adapter_powered ? _('Turn Off') : _('Turn On');
     }
 });
